pkg/controller/servicemesh/webhookca/source.go

```diff
@@ -50,17 +51,19 @@ func (s *SecretCABundleSource) GetCABundle(ctx context.Context, client client.Cl
 			errList = append(errList, err)
 			continue
 		}
-		var ok bool
-		caBundle, ok = secret.Data[entry.Key]
-		if !ok {
-			errList = append(errList, fmt.Errorf(
-				"secret %s does not contain root certificate under key %s",
-				namespacedName,
-				entry.Key,
-			))
-			continue
+		for _, key := range entry.Keys {
+			var ok bool
+			caBundle, ok = secret.Data[key]
+			if !ok {
+				errList = append(errList, fmt.Errorf(
+					"secret %s does not contain root certificate under key %s",
+					namespacedName,
+					key,
+				))
+				continue
+			}
+			break loop
```

Comment:
This (and the previous version) looks more complicated that it is. How about:
```suggestion
			var ok bool
			caBundle, ok = secret.Data[key]
			if ok {
				break mainLoop
			}
			errList = append(errList, fmt.Errorf(
				"secret %s does not contain root certificate under key %s",
				namespacedName,
				key,
			))
```

Actually, instead of `break mainLoop`, this really should just `return caBundle, nil`. I don't see why some people are obsessed with having a single `return` at the end of the function. As in this case, it makes things more convoluted.

Reply:
> Actually, instead of `break mainLoop`, this really should just return `caBundle, nil`

I like this approach, I will do it.

> why some people are obsessed with having a single `return` at the end of the function

It's not my preference. I just modified the existing code with the minimal impact.

Reply:
I know, I wasn't referring to you, but people in the dev community in general.

Reply:
But I honestly think you don't need these changes at all. You could just add a second `SecretNameKeyPair`, as I mentioned above.

Reply:
>  You could just add a second SecretNameKeyPair

Done